bootstrap.py

```diff
@@ -116,16 +117,48 @@ def main(coderoot: str, argv: Sequence[str] | None = None) -> int:
         # as it applies new migrations as well and so would need to ensure
         # the appropriate devservices are running
         proc.run_stream_output(
-            # a new interactive shell is started here so that we get things like updated PATH
-            (shell, "-i", "-e", "-c", "direnv allow ; direnv exec . make bootstrap"),
+            (
+                shell,
+                # interactive shell is used so we can make sure our earlier shellrc
+                # modifications are good
+                "-i",
+                "-e",
+                "-c",
```

Comment:
we should factor out shells entirely -- this won't work on weird shells (and `-i` / etc. opens us up to people's broken rcfiles)

Reply:
will f/u with this!